sdk/avalon_sdk/ethereum/ethereum_listener.py

```diff
@@ -36,127 +31,8 @@ def __init__(self, config):
-        """
-        provider = self._config["ethereum"]["provider"]
-        evt_provider = self._config["ethereum"]["event_provider"]
-
-        if provider == evt_provider:
-            return False
-        else:
-            return True
-
-
-def get_last_read_block():
-    f = None
-    try:
-        f = open("bookmark", 'r')
-        block_num = f.readline()
-        return 0 if block_num == '' else int(block_num)
-    except FileNotFoundError as e:
-        logging.error(e)
-        return 0
-    finally:
-        if f is not None:
-            f.close()
-
-
-def set_last_read_block(block_num):
-    with open("bookmark", 'w') as f:
-        f.seek(0)
-        f.write(str(block_num))
-
-
-def get_new_filter_payload(evt_hash, contract_address):
-    params = {}
-    params["fromBlock"] = "earliest"
-    params["toBlock"] = "latest"
-    params["address"] = contract_address
-    params["topics"] = [[evt_hash]]
-
-    return [params]
-
-
-def parse_jrpc_evt(evt, wo_submitted_hash, wo_completed_hash):
-    """
-    This method parses an event log and converts it to an
-    intelligible format
-    """
-    data = evt["data"]
-    indexed_data = evt["topics"]
-    evt_hash = indexed_data[0]
-    if is_valid_hex_str(evt["blockNumber"]):
-        block_number = int(evt["blockNumber"], 16)
-    else:
-        block_number = evt["blockNumber"]
-
-    if evt_hash == wo_submitted_hash:
-        workOrderRequest, errCode, senderAddress, ver = decode_abi(
-            ["string", "uint256", "address", "bytes4"],
-            bytes.fromhex(data[2:]))  # Truncate preceeding '0x'
-
-        return {"args": {"workOrderRequest": workOrderRequest},
-                "event": "workOrderSubmitted",
-                "blockNumber": block_number}
-
-    elif evt_hash == wo_completed_hash:
-        requesterId, workOrderId,  workOrderStatus, workOrderResponse,\
-            errorCode, version = decode_abi(
-                ["bytes32", "bytes32", "uint256",
-                 "string", "uint256", "bytes4"],
-                bytes.fromhex(data[2:]))
-        # TODO : Extract necessary fields into event to be returned
-        return {"args": {"workOrderId": workOrderId, "version": version,
-                         "workOrderResponse": json.loads(workOrderResponse),
-                         "requesterId": requesterId,
-                         "workOrderStatus": workOrderStatus,
-                         "errorCode": errorCode},
-                "blockNumber": block_number, "event": "workOrderCompleted"}
-
-
-def normalize_event(evt, wo_submitted_hash, wo_completed_hash):
-    """
-    This method reads "jsonrpc" field to detect the event source and format
-    it to give all events a similar structure
-    """
-    try:
-        jrpc = evt["jsonrpc"]
-    except KeyError as e:
-        return evt
-    return parse_jrpc_evt(evt, wo_submitted_hash, wo_completed_hash)
+        
+        return contract.events[event].createFilter(fromBlock=fromBlock)
```

Comment:
Please add a comment about this filter . For example,"setting up filter to get any events from latest block committed in block chain "

Reply:
Done.